### PR TITLE
pthread_spinlock api remaining bsd systems

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/unix/bsd/netbsdlike/openbsd/mod.rs [hunk 1]
```diff
@@ -383,6 +385,13 @@ s! {
         pub dlpi_phdr: *const Elf_Phdr,
         pub dlpi_phnum: Elf_Half,
     }
+
+    // pthread.h
+
+    pub struct __c_anonymous_pthread_spinlock {
```
Comment (semarie) [inline src/unix/bsd/netbsdlike/openbsd/mod.rs:391]: `pthread_spinlock` struct is internal and isn't in public header on purpose. the visible part is only a pointer, nothing else.

it is the same kind of type as `pthread_t` which is defined as `pub type pthread_t = ::uintptr_t` in Rust.

File: src/unix/bsd/netbsdlike/openbsd/mod.rs [hunk 2]
```diff
@@ -1530,6 +1539,12 @@ extern "C" {
     pub fn pthread_main_np() -> ::c_int;
     pub fn pthread_set_name_np(tid: ::pthread_t, name: *const ::c_char);
     pub fn pthread_stackseg_np(thread: ::pthread_t, sinfo: *mut ::stack_t) -> ::c_int;
+    pub fn pthread_spin_init(lock: *mut pthread_spinlock_t, pshared: ::c_int) -> ::c_int;
```
Comment (semarie) [inline src/unix/bsd/netbsdlike/openbsd/mod.rs:1542]: if I don't mess myself, pthread_spin_ functions could be commun with at least NetBSD.

**Review by semarie — CHANGES_REQUESTED** [review]

File: src/unix/bsd/freebsdlike/dragonfly/mod.rs [hunk 3]
```diff
@@ -193,6 +195,10 @@ s! {
     pub struct cpumask_t {
         ary: [u64; 4],
     }
+
+    pub struct __c_anonymous_pthread_spinlock_s {
```
Comment (semarie) [inline src/unix/bsd/freebsdlike/dragonfly/mod.rs:199]: it isn't exactly my area but I think if the OS define a struct as opaque, it should be kept opaque.

`pthread_spinlock_t` is opaque in DragonflyBSD too: [sys/sys/_pthreadtypes.h](https://github.com/DragonFlyBSD/DragonFlyBSD/blob/c163a4d7ee9c6857ee4e04a3a2cbb50c3de29da1/sys/sys/_pthreadtypes.h)

File: libc-test/build.rs [hunk 4]
```diff
@@ -381,6 +381,7 @@ fn test_openbsd(target: &str) {
         "limits.h",
         "link.h",
         "locale.h",
+        "machine/spinlock.h",
```
Comment (semarie) [inline libc-test/build.rs:384]: this definition shouldn't be need now that you don't use `_atomic_lock_t` type.

the tests passes on OpenBSD without it.

**Review by semarie — CHANGES_REQUESTED** [review]

**Comment by semarie** [comment]
it mess to me you reintroduced `__c_anonymous_pthread_spinlock` on OpenBSD while removing it from DragonflyBSD...

**Comment by devnexen** [comment]
ah yes wrong rebasing

**Comment by semarie** [comment]
rebasing still a bit wrong :
- `if ty.starts_with("__c_anonymous_") {` isn't need anymore in `test_openbsd()` function
- pthread_spin_ functions could be merged with NetBSD inside `src/unix/bsd/netbsdlike/mod.rs`

**Comment by devnexen** [comment]
true. Merci beaucoup for your review ;-)

**Review by semarie — APPROVED** [review]
fine with me. thanks

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Thanks! @bors r+

**Pull request merged**